nexus/db-queries/src/db/datastore/fm.rs

```diff
@@ -0,0 +1,757 @@
+        &self,
+        opctx: &OpContext,
+    ) -> Result<Option<(fm::SitrepVersion, Sitrep)>, Error> {
+        let conn = self.pool_connection_authorized(opctx).await?;
+        let version: fm::SitrepVersion =
+            match self.fm_current_sitrep_version_on_conn(&conn).await? {
+                Some(version) => version.into(),
+                None => return Ok(None),
+            };
+        let sitrep = self.fm_sitrep_read_on_conn(version.id, &conn).await?;
+        Ok(Some((version, sitrep)))
+    }
+
+    /// Reads the entire content of the sitrep with the provided ID, if one exists.
+    pub async fn fm_sitrep_read(
+        &self,
+        opctx: &OpContext,
+        id: SitrepUuid,
+    ) -> Result<Sitrep, Error> {
+        opctx.authorize(authz::Action::ListChildren, &authz::FLEET).await?;
+        let conn = self.pool_connection_authorized(opctx).await?;
+        self.fm_sitrep_read_on_conn(id, &conn).await
+    }
+
+    async fn fm_sitrep_read_on_conn(
+        &self,
+        id: SitrepUuid,
+        conn: &async_bb8_diesel::Connection<DbConnection>,
+    ) -> Result<Sitrep, Error> {
+        let metadata =
+            self.fm_sitrep_metadata_read_on_conn(id, &conn).await?.into();
+
+        // TODO(eliza): this is where we would read all the other sitrep data,
+        // if there was any.
+
+        Ok(Sitrep { metadata })
+    }
+
+    /// Insert the provided [`Sitrep`] into the database, and attempt to mark it
+    /// as the current sitrep.
+    ///
+    /// If the sitrep's parent is not the current sitrep, the new sitrep is not
+    /// added to the sitrep history, and an error is returned. See [this
+    /// section](https://rfd.shared.oxide.computer/rfd/0603#_creating_sitreps)
+    /// in RFD 603 for details.
+    ///
+    /// # Returns
+    ///
+    /// - `Ok(())` if the new sitrep was both successfully inserted *and* added
+    ///   to the sitrep history as the current sitrep.
+    ///
+    /// - `Err(`[`InsertSitrepError::ParentNotCurrent`]`)` if the sitrep's
+    ///    `parent_sitrep_id` is not the current sitrep, indicating that it was
+    ///    generated based on out of date inputs.
+    ///
+    ///    This error indicates that the sitrep is orphaned and should be
+    ///    deleted. It is out of date, and another sitrep has already been
+    ///    generated based on the same inputs.
+    ///
+    /// - `Err(`[`InsertSitrepError::Other`]`)` if another error occurred while
+    ///    inserting the sitrep.
+    pub async fn fm_sitrep_insert(
+        &self,
+        opctx: &OpContext,
+        sitrep: &Sitrep,
+    ) -> Result<(), InsertSitrepError> {
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        // TODO(eliza): there should probably be an authz object for the fm sitrep?
+        opctx.authorize(authz::Action::Modify, &authz::FLEET).await?;
+
+        // Create the sitrep metadata record.
+        diesel::insert_into(sitrep_dsl::fm_sitrep)
```

Comment:
As written, if we crash after this "insert_into" but before the "InsetSitrepVersionQuery", we'll also leak some rows.

To confirm: Nothing is deleting, nor attempting to delete, orphaned/partially written sitreps in this PR, correct?

Reply:
That's correct, I was going to do that next.

Reply:
#9335 adds a background task with the grisly duty of hunting down and killing orphans.